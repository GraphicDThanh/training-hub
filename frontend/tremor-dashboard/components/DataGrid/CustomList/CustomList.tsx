import { OrderProduct } from "@/types";

interface CustomListProps {
  products: OrderProduct[];
  className?: string;
}

const CustomList = ({ products, className = "w-[180px]" }: CustomListProps) => (
  <>
    {!!products.length &&
      products.map(product => (
        <p
          data-testid="order-product"
          key={product.id}
          className={`text-tertiary text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] truncate order-product first:pt-0 last:pt-5 first:pb-0 last:pb-0 border-0 dark:border-grayish border-t border-gray-100 first:mb-5 first:border-0 ${className}`}>
          {product.name}
        </p>
      ))}
  </>
);

export default CustomList;
