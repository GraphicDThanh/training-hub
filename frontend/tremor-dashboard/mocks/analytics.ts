export const SALES_BY_COUNTRY = [
  {
    id: "us",
    flag: "/assets/images/flags/US.webp",
    country: "United State",
    sales: 2500,
    value: 230900,
    bounce: 29.9,
  },
  {
    id: "ge",
    flag: "/assets/images/flags/DE.webp",
    country: "Germany",
    sales: 3900,
    value: 440000,
    bounce: 40.22,
  },
  {
    id: "gb",
    flag: "/assets/images/flags/GB.webp",
    country: "Great Britain",
    sales: 1400,
    value: 190700,
    bounce: 23.44,
  },
  {
    id: "br",
    flag: "/assets/images/flags/BR.webp",
    country: "Brasil",
    sales: 562,
    value: 143960,
    bounce: 32.14,
  },
  {
    id: "au",
    flag: "/assets/images/flags/AU.webp",
    country: "Australia",
    sales: 400,
    value: 143960,
    bounce: 56.83,
  },
];

export const STATISTICAL_DATA = [
  {
    id: "1",
    type: "Bookings",
    amount: 281,
    amountChange: 55,
    duration: "than last week",
    amountChangeType: 0,
  },
  {
    id: "2",
    type: "Today's Users",
    amount: 2300,
    amountChange: 3,
    duration: "than last month",
    amountChangeType: 0,
  },
  {
    id: "3",
    type: "Revenue",
    amount: 34,
    amountChange: 1,
    duration: "than yesterday",
    amountChangeType: 0,
  },
  {
    id: "4",
    type: "Followers",
    amount: 91,
    amountChange: 0,
    duration: "Just updated",
    amountChangeType: 0,
  },
];

export const ANALYTIC_INFO = [
  {
    id: "1",
    photo: "/assets/images/analytics/analytics-info-1.webp",
    name: "Cozy 5 Stars Apartment",
    description: `The place is close to Barceloneta Beach and bus stop just 2 min by walk and near to "Navig" where you can enjoy the main night life in Barcelona.`,
    price: 899,
    location: "Barcelona, Spain",
  },
  {
    id: "2",
    photo: "/assets/images/analytics/analytics-info-2.webp",
    name: "Office Studio",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Navig" where you can enjoy the night life in London, UK.`,
    price: 1.119,
    location: "London, UK",
  },
  {
    id: "3",
    photo: "/assets/images/analytics/analytics-info-3.webp",
    name: "Beautiful Castle",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Navig" where you can enjoy the main night life in Milan.`,
    price: 459,
    location: "Milan, Italy",
  },
];

export const WEBSITE_CHART = {
  id: "view-campaign",
  display: "Website Views",
  desc: "Last Campaign Performance",
  modified: "campaign sent 2 days ago",
  data: [
    {
      description: "M",
      sales: 50,
    },
    {
      description: "T",
      sales: 20,
    },
    {
      description: "W",
      sales: 10,
    },
    {
      description: "T",
      sales: 22,
    },
    {
      description: "F",
      sales: 50,
    },
    {
      description: "S",
      sales: 10,
    },
    {
      description: "S",
      sales: 40,
    },
  ],
};

export const LINE_CHART_DATA = [
  {
    id: "sale",
    display: "Daily Sales",
    descValue: "(+15%)",
    desc: "increase in today sales.",
    modified: "updated 4 min ago",
    data: [
      {
        date: "Apr",
        "Desktop apps": 50,
      },
      {
        date: "May",
        "Desktop apps": 40,
      },
      {
        date: "Jun",
        "Desktop apps": 300,
      },
      {
        date: "Jul",
        "Desktop apps": 220,
      },
      {
        date: "Aug",
        "Desktop apps": 500,
      },
      {
        date: "Sep",
        "Desktop apps": 250,
      },
      {
        date: "Oct",
        "Desktop apps": 400,
      },
      {
        date: "Nov",
        "Desktop apps": 230,
      },
      {
        date: "Dec",
        "Desktop apps": 500,
      },
    ],
  },
  {
    id: "performance",
    display: "Completed Tasks",
    descValue: "(+15%)",
    desc: "Last Campaign Performance",
    modified: "just updated",
    data: [
      {
        date: "Apr",
        "Mobile apps": 50,
      },
      {
        date: "May",
        "Mobile apps": 40,
      },
      {
        date: "Jun",
        "Mobile apps": 300,
      },
      {
        date: "Jul",
        "Mobile apps": 320,
      },
      {
        date: "Aug",
        "Mobile apps": 500,
      },
      {
        date: "Sep",
        "Mobile apps": 350,
      },
      {
        date: "Oct",
        "Mobile apps": 200,
      },
      {
        date: "Nov",
        "Mobile apps": 230,
      },
      {
        date: "Dec",
        "Mobile apps": 500,
      },
    ],
  },
];

export const ANALITICS_DATA = {
  apartment_statistic: ANALYTIC_INFO,
  daily_sale_statistic: LINE_CHART_DATA[0],
  performance_statistic: LINE_CHART_DATA[1],
  sale_by_country: SALES_BY_COUNTRY,
  sale_statistical: STATISTICAL_DATA,
  web_statistic: WEBSITE_CHART,
};

export const EMPTY_ANALITICS_DATA = {
  apartment_statistic: [],
  daily_sale_statistic: {},
  performance_statistic: {},
  sale_by_country: [],
  sale_statistical: [],
  web_statistic: {},
};
