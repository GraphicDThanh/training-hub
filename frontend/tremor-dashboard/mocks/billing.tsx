// Constants
import { AGGREGATION_TYPE } from "@/constants";
import { BillingInfoData } from "@/types";

export const MOCK_SALARY_DATA = [
  {
    type: AGGREGATION_TYPE.SALARY,
    value: 2000,
  },
  {
    type: AGGREGATION_TYPE.PAYPAL,
    value: 45500,
  },
];

export const MOCK_BILLING_CARD = {
  expire: "11/24",
  cardNumber: "4562 1122 4594 7866",
  holderFullName: "Jack Peterson",
  cardLast4Digit: "7866",
};

export const MOCK_BILLING_INFO: BillingInfoData = {
  id: "1",
  ownerName: "Oliver Liam",
  companyName: "Viking Burrito",
  cardLast4Digit: "1234",
  email: "oliver@burrito.com",
  vat: "FRB1235476",
};

export const MOCK_BILLING: BillingInfoData[] = [
  {
    id: "1",
    ownerName: "Oliver Liam",
    companyName: "Viking Burrito",
    cardLast4Digit: "1234",
    email: "oliver@burrito.com",
    vat: "FRB1235476",
  },
  {
    id: "2",
    ownerName: "Oliver Liam",
    companyName: "Viking Burrito",
    cardLast4Digit: "1234",
    email: "oliver@burrito.com",
    vat: "FRB1235476",
  },
  {
    id: "3",
    ownerName: "Oliver Liam",
    companyName: "Viking Burrito",
    cardLast4Digit: "1234",
    email: "oliver@burrito.com",
    vat: "FRB1235476",
  },
];
