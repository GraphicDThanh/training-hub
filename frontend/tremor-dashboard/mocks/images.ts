export const MOCK_URL_AVATAR =
  "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/product-details-1.f79d1e89.jpg";

export const MOCK_IMAGE_UPLOAD =
  "https://i.ibb.co/2WyPpQQ/0daa97f4b0dbe9a9686984982977da86-1.png";
