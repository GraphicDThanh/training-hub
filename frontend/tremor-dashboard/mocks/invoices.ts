import { Invoice } from "@/types";

export const MOCK_INVOICE_DATA = {
  id: 230220,
  createdAt: "06/03/2019",
  dueAt: "11/03/2019",
  bankInfo: {
    address: "St. Independence Embankment, 050105",
    city: "Bucharest",
    state: "Romania",
    phone: "+4 (074) 1090873",
  },
  customer: {
    fullName: "John Doe",
    address: "4006 Locust View Drive",
    city: "San Francisco",
    stateCode: "CA",
    state: "California",
  },
  products: [
    {
      id: 1,
      productName: "Premium Support",
      quantity: 1,
      price: 9,
    },
    {
      id: 2,
      productName: "Metro Dashboard",
      quantity: 3,
      price: 100,
    },
    {
      id: 3,
      productName: "Parts for service",
      quantity: 2,
      price: 10,
    },
  ],
  totalCost: 345,
};

export const MOCK_INVOICES: Invoice[] = [
  {
    id: 804322,
    createdAt: "March, 21, 2021",
    invoicePrefix: "AR",
    totalCost: 180,
  },
  {
    id: 8908543,
    createdAt: "Apr, 22, 2022",
    invoicePrefix: "AR",
    totalCost: 180,
  },
  {
    id: 415645,
    createdAt: "May, 23, 2023",
    invoicePrefix: "MS",
    totalCost: 180,
  },
  {
    id: 103578,
    createdAt: "Jun, 24, 2024",
    invoicePrefix: "QW",
    totalCost: 180,
  },
  {
    id: 126749,
    createdAt: "Jul, 25, 2025",
    invoicePrefix: "RV",
    totalCost: 180,
  },
];

export const MOCK_INVOICE_HEADER = {
  addressBank: "St. Independence Embankment, 050105",
  cityBank: "Bucharest",
  stateBank: "Romania",
  phoneBank: "+4 (074) 1090873",
  fullName: "John Doe",
  addressCustomer: "4006 Locust View Drive",
  cityCustomer: "San Francisco",
  stateCode: "CA",
  stateCustomer: "California",
};

//addressBank: string;
//cityBank: string;
//stateBank: string;
//phoneBank: string;
//fullName: string;
//addressCustomer: string;
//cityCustomer: string;
//stateCode: string;
//stateCustomer: string;

//id: number;
//createdAt: string;
//orderCode: string; }> & {
// id?: number | undefined;
//createdAt?: string | undefined;
//orderCode?: string | undefined;
