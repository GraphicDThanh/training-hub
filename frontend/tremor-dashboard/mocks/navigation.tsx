// Icons
import { IoPersonSharp, IoReceiptOutline } from "react-icons/io5";

// Types
import { INavigation } from "@/types";

export const NAVIGATION_SETTING: INavigation[] = [
  {
    icon: (
      <IoPersonSharp className="text-primary dark:text-white dark!:hover:text-black" />
    ),
    label: "profile",
    href: "#profile",
  },
  {
    icon: (
      <IoReceiptOutline className="text-primary dark:text-white dark!:hover:text-black" />
    ),
    label: "basic info",
    href: "#basicInfo",
  },
];
