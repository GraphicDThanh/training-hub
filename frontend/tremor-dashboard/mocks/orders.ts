// Types
import { Order, TrackOrderData } from "@/types";

export const MOCK_ORDERS: Order[] = [
  {
    id: 10425,
    createdAt: "2023-11-01T10:20:00+00:00",
    status: 1,
    customer: {
      id: 15,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 44.9,
    products: [
      {
        id: 6,
        name: "Phone Case Pink",
        count: 2,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 90,
      },
    ],
  },
  {
    id: 12345,
    createdAt: "2023-11-01T03:20:00+00:00",
    status: 0,
    customer: {
      id: 1,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 140.2,
    products: [
      {
        id: 12,
        name: "Office Papers",
        count: 1,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 77,
      },
    ],
  },
  {
    id: 12350,
    createdAt: "2023-11-01T11:03:00+00:00",
    status: 0,
    customer: {
      id: 9,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 112.5,
    products: [
      {
        id: 10,
        name: "Water Bottle India",
        count: 3,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 7,
      },
    ],
  },
  {
    id: 12356,
    createdAt: "2023-11-01T09:20:00+00:00",
    status: 2,
    customer: {
      id: 13,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 23.9,
    products: [
      {
        id: 10,
        name: "Water Bottle India",
        count: 3,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 34.35,
      },
    ],
  },
  {
    id: 12352,
    createdAt: "2023-11-01T16:05:00+00:00",
    status: 0,
    customer: {
      id: 16,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 350,
    products: [
      {
        id: 10,
        name: "Water Bottle India",
        count: 3,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 71,
      },
    ],
  },
];

export const MOCK_INVOICE_ORDER_HEADER = {
  id: 241342,
  createdAt: "2024-01-11T01:48:52Z",
  orderCode: "KF332",
};

export const MOCK_ORDER_SUMMARY = {
  productPrice: 90,
  delivery: 14,
  taxes: 1.95,
};

export const MOCK_TRACK_ORDER: TrackOrderData = {
  id: "#1832412",
  generateOrderAt: "2024-01-11T02:48:52Z",
  deliveredAt: "2024-01-11T02:48:52Z",
  transmittedToCourierAt: "2024-01-11T02:48:52Z",
  generateOrderId: "2024-01-11T02:48:52Z",
};

export const MOCK_ORDER_CONTACT = {
  name: "Leather Wallet",
  url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
  date: 18,
};
