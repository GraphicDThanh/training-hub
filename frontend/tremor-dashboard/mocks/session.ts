import { User } from "@/types";

import * as SessionContext from "@/context/session";

export const mockUseSession = (user: User | null = null) => {
  jest.spyOn(SessionContext, "useSession").mockImplementation(() => ({
    user,
    setUser: () => {},
  }));
};
