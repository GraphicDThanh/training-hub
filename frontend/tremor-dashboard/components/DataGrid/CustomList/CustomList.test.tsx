import { render } from "@testing-library/react";

// Components
import CustomList from "./CustomList";

// Mocks
import { MOCK_PRODUCTS_SHORTER_FORM } from "@/mocks";

import { OrderProduct } from "@/types";

describe("CustomList component", () => {
  it("should render item list correctly", () => {
    const { container } = render(
      <CustomList products={MOCK_PRODUCTS_SHORTER_FORM} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("should test how many items rendered", () => {
    const { getAllByTestId } = render(
      <CustomList products={MOCK_PRODUCTS_SHORTER_FORM} />,
    );
    const lengthList = MOCK_PRODUCTS_SHORTER_FORM.length;
    expect(getAllByTestId("order-product")).toHaveLength(lengthList);
  });

  it("should not render any items if empty products", () => {
    const { queryAllByTestId } = render(
      <CustomList products={[] as OrderProduct[]} />,
    );
    expect(queryAllByTestId("order-product")).toHaveLength(0);
  });
});
