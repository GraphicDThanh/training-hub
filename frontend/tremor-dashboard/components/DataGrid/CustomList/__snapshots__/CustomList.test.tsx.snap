// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomList component should render item list correctly 1`] = `
<div>
  <p
    class="text-tertiary text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] truncate order-product first:pt-0 last:pt-5 first:pb-0 last:pb-0 border-0 dark:border-grayish border-t border-gray-100 first:mb-5 first:border-0 w-[180px]"
    data-testid="order-product"
  >
    Premium Support
  </p>
  <p
    class="text-tertiary text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] truncate order-product first:pt-0 last:pt-5 first:pb-0 last:pb-0 border-0 dark:border-grayish border-t border-gray-100 first:mb-5 first:border-0 w-[180px]"
    data-testid="order-product"
  >
    Metro Dashboard
  </p>
  <p
    class="text-tertiary text-xs dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] truncate order-product first:pt-0 last:pt-5 first:pb-0 last:pb-0 border-0 dark:border-grayish border-t border-gray-100 first:mb-5 first:border-0 w-[180px]"
    data-testid="order-product"
  >
    Parts for service
  </p>
</div>
`;
